File: app/src/main/java/com/jerson/hcdc_portal/ui/adapter/BuildingDetailsAdapter.java
package com.jerson.hcdc_portal.ui.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.jerson.hcdc_portal.databinding.ItemContainerBuildingFloorBinding;
import com.jerson.hcdc_portal.listener.OnClickListener;
import com.jerson.hcdc_portal.model.BuildingModel;

import java.util.List;

public class BuildingDetailsAdapter extends RecyclerView.Adapter<BuildingDetailsAdapter.ViewHolder> {
    private List<BuildingModel.Floor> list;
    private OnClickListener<Integer> listener;

    public BuildingDetailsAdapter(List<BuildingModel.Floor> list, OnClickListener<Integer> listener) {
        this.list = list;
        this.listener = listener;
    }

    @NonNull
    @Override
    public BuildingDetailsAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(ItemContainerBuildingFloorBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull BuildingDetailsAdapter.ViewHolder holder, int position) {
        BuildingModel.Floor item = list.get(position);
        holder.binding.floor.setText(getFloorText(item.getFloor_no()));
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        private final ItemContainerBuildingFloorBinding binding;

        public ViewHolder(@NonNull ItemContainerBuildingFloorBinding itemView) {
            super(itemView.getRoot());
            this.binding = itemView;

            binding.getRoot().setOnClickListener(v -> {
                listener.onItemClick(getAdapterPosition());
            });
        }
    }

    String getFloorText(int i) {
        String s;
        switch (i) {
            case 1:
                s = "1st";
                break;
            case 2:
                s = "2nd";
                break;
            case 3:
                s = "3rd";
                break;
            default:
                s = i + "th";
        }

        return s;
    }
}

File: app/src/main/java/com/jerson/hcdc_portal/presentation/building/adapter/BuildingDetailsAdapter.kt
package com.jerson.hcdc_portal.presentation.building.adapter

import android.content.res.ColorStateList
import android.graphics.Color
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.jerson.hcdc_portal.databinding.ItemContainerBuildingFloorBinding
import com.jerson.hcdc_portal.domain.model.Floor

class BuildingDetailsAdapter(private val list:List<Floor>,private val itemCallback:(Floor)-> Unit): RecyclerView.Adapter<BuildingDetailsAdapter.ViewHolder>() {
    private var prevSelectedPos = -1
    inner class ViewHolder(private val binding: ItemContainerBuildingFloorBinding):RecyclerView.ViewHolder(binding.root){
        fun bind(item:Floor,pos:Int){
            binding.floor.text = item.floor_no.toString()
            binding.card.setOnClickListener{
                itemSelectedIndicator(binding.viewSelected,pos)
                itemCallback(item)
            }

            binding.viewSelected.backgroundTintList =
                if (pos == prevSelectedPos) ColorStateList.valueOf(Color.GREEN)
                else ColorStateList.valueOf(Color.TRANSPARENT)
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(
            ItemContainerBuildingFloorBinding.inflate(
                LayoutInflater.from(parent.context),
                parent,
                false
            )
        )
    }

    override fun getItemCount(): Int {
        return list.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = list[position]
        holder.bind(item,position)

    }

    private fun itemSelectedIndicator(view: View, curPos: Int) {
        if (prevSelectedPos != RecyclerView.NO_POSITION) {
            notifyItemChanged(prevSelectedPos)
        }
        prevSelectedPos = curPos
        notifyItemChanged(curPos)
        view.backgroundTintList =  ColorStateList.valueOf(Color.GREEN)
    }
}